pbkdf2-check.c

```diff
@@ -108,15 +108,35 @@ int pbkdf2_check(char *password, char *hash)
 		return (FALSE);
 	}
 
+#ifdef RAW_SALT
+	char *rawSalt;
+
+	if ((rawSalt = malloc(strlen(salt) + 1)) == NULL) {
+		fprintf(stderr, "Out of memory\n");
+		return FALSE;
+	}
+
+	saltlen = base64_decode(salt, rawSalt);
+	if (saltlen < 1) {
+		return (FALSE);
+	}
+
+	free(salt);
+	salt = strdup(rawSalt);
```

Comment:
This has been fixed in https://github.com/jpmens/mosquitto-auth-plug/pull/262/commits/49eaaf9e85ebc9d0e8c1075c58f4c0cd33b6afd0.